examples/analysis/lib/strict_modes.dart

```diff
@@ -0,0 +1,32 @@
+// ignore_for_file: dead_code, prefer_typing_uninitialized_variables
+
+import 'package:examples_util/ellipsis.dart';
+
+String strictCasts() {
+  // #docregion strict-casts
+  dynamic o = ellipsis<String>();
+  // ignore: stable, beta, dev, invalid_assignment
+  String s = o; // Implicit downcast
+  String s2 = s.substring(1);
+  // #enddocregion strict-casts
+  return s2;
+}
+
+void strictInference() {
+  // #docregion strict-inference
+  // ignore: stable, beta, dev, inference_failure_on_uninitialized_variable
+  var dynamicValue;
```

Comment:
The collection literal examples might be more interesting, and I think are more common errors than this example.

https://github.com/dart-lang/language/blob/master/resources/type-system/strict-inference.md#collection-literal

Reply:
I had a little trouble with this as I see the potential errors here having a lot of similarity to `strict-raw-types` but let me know what you think of the newest example?